examples/circuitrelay.nim

```diff
@@ -26,15 +26,18 @@ proc main() {.async.} =
   let customProtoCodec = "/test"
   var proto = new LPProtocol
   proto.codec = customProtoCodec
-  proto.handler = proc(conn: Connection, proto: string) {.async.} =
-    var msg = string.fromBytes(await conn.readLp(1024))
-    echo "1 - Dst Received: ", msg
-    assert "test1" == msg
-    await conn.writeLp("test2")
-    msg = string.fromBytes(await conn.readLp(1024))
-    echo "2 - Dst Received: ", msg
-    assert "test3" == msg
-    await conn.writeLp("test4")
+  proto.handler = proc(conn: Connection, proto: string) {.async: (raises: []).} =
+    try:
+      var msg = string.fromBytes(await conn.readLp(1024))
+      echo "1 - Dst Received: ", msg
+      assert "test1" == msg
+      await conn.writeLp("test2")
+      msg = string.fromBytes(await conn.readLp(1024))
+      echo "2 - Dst Received: ", msg
+      assert "test3" == msg
+      await conn.writeLp("test4")
+    except:
```

Comment:
we avoid `except:` as it hides defect and is undefined behavior - always `CatchableError` or a more specific exception

Reply:
`except:` and `except CatchableError as e:` should be the same thing, right? 

i have used `except:` when i don't need exception variable `e`, for example: 
- in tests, if exceptions happens, test should fail no matter what
- in examples, if exceptions happens, should print to stdout (it is way for handling errors in examples)

Reply:
> should be the same thing, right?

Reply:
`except:` == `except Exception:` which is different - it also catches `Defect` _sometimes_ which leads to problems.

Reply:
`except CatchableError:` is the canonical way to catch everything, when you don't need `e` (rather unfortunate, but a backwards-compatibility issue in nim)